Tests/HealthKitOnFHIRTests/HKCategorySampleTests.swift

```diff
@@ -0,0 +1,472 @@
+                value: value.rawValue
+            )
+
+            XCTAssertEqual(
+                observation.code.coding?.first,
+                createCategoryCoding(
+                    categoryType: HKCategoryType(.ovulationTestResult).description,
+                    display: "Ovulation Test Result"
+                )
+            )
+            XCTAssertEqual(
+                observation.value, .string(value.description.asFHIRStringPrimitive())
+            )
+        }
+    }
+
+    func testContraceptive() throws {
+        let values: [HKCategoryValueContraceptive] = [.unspecified, .implant, .injection, .intrauterineDevice, .intravaginalRing, .oral, .patch]
+
+        for value in values {
+            let observation = try createObservationFrom(
+                type: HKCategoryType(.contraceptive),
+                value: value.rawValue
+            )
+
+            XCTAssertEqual(
+                observation.code.coding?.first,
+                createCategoryCoding(
+                    categoryType: HKCategoryType(.contraceptive).description,
+                    display: "Contraceptive"
+                )
+            )
+            XCTAssertEqual(observation.value, .string(value.description.asFHIRStringPrimitive()))
+        }
+    }
+
+    func testSleepAnalysis() throws {
+        let values: [HKCategoryValueSleepAnalysis] = [.inBed, .asleepUnspecified, .awake, .asleepCore, .asleepDeep, .asleepREM]
+
+        for value in values {
+            let observation = try createObservationFrom(
+                type: HKCategoryType(.sleepAnalysis),
+                value: value.rawValue
+            )
+
+            XCTAssertEqual(
+                observation.code.coding?.first,
+                createCategoryCoding(
+                    categoryType: HKCategoryType(.sleepAnalysis).description,
+                    display: "Sleep Analysis"
+                )
+            )
+            XCTAssertEqual(observation.value, .string(value.description.asFHIRStringPrimitive()))
+        }
+    }
+
+    func testAppetiteChanges() throws {
+        let values: [HKCategoryValueAppetiteChanges] = [.unspecified, .noChange, .decreased, .increased]
+
+        for value in values {
+            let observation = try createObservationFrom(
+                type: HKCategoryType(.appetiteChanges),
+                value: value.rawValue
+            )
+
+            XCTAssertEqual(
+                observation.code.coding?.first,
+                createCategoryCoding(
+                    categoryType: HKCategoryType(.appetiteChanges).description,
+                    display: "Appetite Changes"
+                )
+            )
+            XCTAssertEqual(observation.value, .string(value.description.asFHIRStringPrimitive()))
+        }
+    }
+
+    func testEnvironmentalAudioExposureEvent() throws {
+        let observation = try createObservationFrom(
+            type: HKCategoryType(.audioExposureEvent),
```

Comment:
To remove the warning:
```suggestion
            type: HKCategoryType(.environmentalAudioExposureEvent),
```

Reply:
I have tried this, but the corresponding string value of this type "HKCategoryTypeIdentifierEnvironmentalAudioExposureEvent" is failing validation when being read in from the JSON mapping.

`HealthKitOnFHIR/HKSampleMapping.swift:58: Fatal error: HKCategoryType for the String value HKCategoryTypeIdentifierEnvironmentalAudioExposureEvent does not exist.`

Reply:
Thank you, I will take a look at the branch, thanks for reproducing the issue! 👍

Reply:
I have reproduced this: The non-deprecated `HKCategoryType(.environmentalAudioExposureEvent)`'s string value is still `HKCategoryTypeIdentifierAudioExposureEvent`. Therefore you can still use that string key in the JSON and use the non-deprecated enum value in the JSON configuration. I have pushed this to the other branch and will create a PR for this to this branch 👍

Reply:
Thanks. That is a bit confusing - perhaps we should report this as an issue to the HealthKit team.

Reply:
Have merged this in, thanks!